packages/react-api/src/Api.tsx

```diff
@@ -82,13 +84,14 @@ async function getInjectedAccounts (injectedPromise: Promise<InjectedExtension[]
 
     const accounts = await web3Accounts();
 
-    return accounts.map(({ address, meta }, whenCreated): InjectedAccountExt => ({
+    return accounts.map(({ address, meta, type }, whenCreated): InjectedAccountExt => ({
```

Comment:
On line 85 above it should explicitly ask for the `ethereum` types, when the flag is passed. (Comments in extension on this - I'm guessing this is based on the `isEthereum` flag)

Reply:
EDIT: Since we do filter on the type in the keyring anyway, it is most probably not needed, since we do have the same effect. (Probably just more descriptive always being explicit, i.e. no need to even pass these through unless needed)

Reply:
So what do you think I should do? Leave as is?

Reply:
Ahhhh.... ok this links through to this comment as well https://github.com/polkadot-js/extension/pull/566#discussion_r689374988

In the keyring itself, with the injected load, we don't actually check the type at all, e.g. https://github.com/polkadot-js/ui/blob/master/packages/ui-keyring/src/Keyring.ts#L249-L260 - we just inject what is there. The best place to add a solution is either -

- in the extension, i.e. add an extra flag (may not be the prettiest, as described in the linked comment it is kinda ugly)
- on the keyring itself, i.e. filtering by type there (just need to follow there again, you did do some type filtering already there, maybe I'm missing something in the link)

Would not add a solution inside the apps UI since there are a lot of extension-dapp users, and don't want them to have to make all the same changes.

Anyway, we can test this once all ok.

Reply:
I added the filter in the extension dapp (in web3accounts), so I will pass down the type flag here